### PR TITLE
Allow extracting a &Collector from a Guard

### PR DESCRIPTION
**Pull request by Amanieu**
This is needed for https://github.com/crossbeam-rs/crossbeam-skiplist/issues/4

### CODE REVIEW

File: src/internal.rs [hunk 1]
```diff
@@ -188,28 +187,34 @@ impl Local {
     const PINNINGS_BETWEEN_COLLECT: usize = 128;
 
     /// Registers a new `Local` in the provided `Global`.
-    pub fn register(global: &Arc<Global>) -> Handle {
+    pub fn register(collector: &Collector) -> Handle {
```
Comment (ghost) [inline src/internal.rs:190]: This is a really minor nit, but would it make more sense to accept a collector by value rather than reference here?

**Review by ghost — APPROVED** [review]
Looks good!

Do you think it'd make sense to also expose the following methods (maybe later, not necessarily now)?

* `Handle::collector(&self) -> &Collector`
* `Guard::handle(&self) -> Option<&Handle>`

File: src/internal.rs (hunk 1 above)
Comment (jeehoonkang) [reply src/internal.rs:190]: I changed it because of line 203: we need to have `collector` after making `local`.

**Review by jeehoonkang — APPROVED** [review]
Last year, we thought that maybe the internal API will be useful in building custom collectors.  Basically that's the reason why there are `internal.rs` and `collector.rs`.  But we couldn't find a good way to do that.

To my eyes, this PR basically says there's no longer the internal API: they exists only as the implementation details of handles and collectors.  I think it's a right direction for the current time being.

**Comment by Amanieu** [comment]
I added `default_collector` to get a reference to the global collector, and `Handle::collector`.

**Pull request merged**